monitoring/uss_qualifier/run_locally.sh

```diff
@@ -25,7 +25,7 @@ monitoring/build.sh || exit 1
 CONFIG_NAME="${1:-configurations.dev.local_test}"
 CONFIG_FLAG="--config ${CONFIG_NAME}"
 
-AUTH_SPEC='DummyOAuth(http://host.docker.internal:8085/token,uss_qualifier)'
+AUTH_SPEC='DummyOAuth(http://host.docker.internal:8085/token,uss_qualifier, message_signing=UFT(signing_key_path= /var/test-certs/messagesigning/mock_faa_priv.pem, cert_url = http://host.docker.internal:8077/mock/msgsigning/.well-known/uas-traffic-management/pub.der))'
```

Comment:
It should be both. Mock USS needs the auth spec so it signs it's responses/validate incoming signed http requests, and uss_qualifier needs it to sign it's outgoing requests

Reply:
Hmm, I think we're going to need some pretty articulate documentation about this.

![Message signing architecture diagram](https://github.com/interuss/dss/blob/master/monitoring/uss_qualifier/suites/faa/uft/design/InterUss_Test_Harness_With_Message_Signing.png?raw=true)

The architecture diagram doesn't show any connections from uss_qualifier that need message signing.  Logging control/retrieval with mock_uss is performed via a custom interface that is internal to InterUSS.  Instructions to inject operations is via the InterUSS flight planning/SCD automated testing interface which doesn't need message signing as it's not part of the UTM ecosystem under test.

Thinking harder about what this suite will do, however, I realize that part of the F3548 suite involves uss_qualifier acting as a USS to double-check the operational intent a USS claims to have created to serve a user flight intent requested through the flight planning/SCD automated testing interface.  This means uss_qualifier will call a USS's operational intent details endpoint directly, and that endpoint is within the ASTM standard, thus part of the UTM ecosystem under test, thus possibly in scope for message signing.  I think this introduces some issues.

Currently, uss_qualifier does not host any endpoints -- it makes outgoing calls only, on behalf of the USS running it.  By adding this line, we're effectively saying that the port-8077 mock USS is an essential component of the USS running uss_qualifier.  This is the first time we've made that assumption, I don't think it's documented anywhere, and it's surprising.  There are a number of implications: first, the port-8077 mock USS must have the same participant ID as the USS running uss_qualifier (because they're components of the same USS).  In a real deployment, I don't think this would be the case since a real USS running uss_qualifier would host their cert somewhere other than a mock USS instance, so we can't programmatically check that the participant IDs match.  Instead, we'd need to write some very clear documentation describing the situation so users can understand what is being assumed and what is required.  Second, we shouldn't even use the port-8077 mock_uss's cert_url as the cert_url for the USS running uss_qualifier even in local/dev tests because we're eventually going to [intentionally mangle](https://github.com/interuss/dss/tree/master/monitoring/uss_qualifier/suites/faa/uft/design#negative-tests--) the certs of the port-8077 mock_uss as part of the test.  The USS running uss_qualifier should always work properly throughout the test, so we shouldn't/can't mangle their certs if they need to use message signing.

This is a challenging problem.  I'd suggest that the simplest solution is probably to just not have uss_qualifier do message signing.  This will work with UFT because the message signing requirements have been relaxed to accept and log messages with no signatures to accommodate test tools, and uss_qualifier is indeed a test tool.  If that solution isn't acceptable, I think we'll need to update the design doc to think about this issue more carefully.  I suspect the solution will probably be that message signing will not be able to use the existing F3548 suite as currently envisioned.  The most obvious place where this won't work is in the negative cases -- behavior expectations are different between happy-path-F3548 and negative-case-message-signing.  When port-8077 mock_uss is set to have mangled message signing certs, the test should pass if the intent to create a flight fails (since message signing won't work properly) whereas in plain F3548, the test should pass if the intent to create a flight succeeds.  In that case, we'll simply need separate test scenarios for message signing, and in that case, there would be no need to check the validity of operational intent details since there's no message signing requirement that check would cause to fail, therefore uss_qualifier would not need to sign its messages.  I think this matches the original high-level vision for message signing testing which was that mock_uss is the tooling which manipulates and measures message signing capabilities (rather than uss_qualifier).